src/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs

```diff
@@ -0,0 +1,651 @@
+        {
+            if (value is string)
+            {
+                // Hack, Only when GetCultureName returns culture.Name, we use CultureInfoMapper 
+                // (Since CultureInfoMapper will transfer Culture.DisplayName to Culture.Name).
+                // Otherwise, we just keep the value unchange.
+                string text = (string)value;
+                if (GetCultureName(CultureInfo.InvariantCulture).Equals(""))
+                {
+                    text = CultureInfoMapper.GetCultureInfoName((string)value);
+                }
+                CultureInfo retVal = null;
+
+                CultureInfo currentUICulture = CultureInfo.CurrentUICulture;
+
+                if (culture != null && culture.Equals(CultureInfo.InvariantCulture))
+                {
+                    CultureInfo.CurrentUICulture = culture;
+                }
+
+                try
+                {
+                    // Look for the default culture info.
+                    //
+                    if (text == null || text.Length == 0 || string.Compare(text, DefaultCultureString, StringComparison.Ordinal) == 0)
+                    {
+                        retVal = CultureInfo.InvariantCulture;
+                    }
+
+                    // Now look in our set of installed cultures.
+                    //
+                    if (retVal == null)
+                    {
+                        ICollection values = GetStandardValues(context);
+                        IEnumerator e = values.GetEnumerator();
+                        while (e.MoveNext())
+                        {
+                            CultureInfo info = (CultureInfo)e.Current;
+                            if (info != null && string.Compare(GetCultureName(info), text, StringComparison.Ordinal) == 0)
+                            {
+                                retVal = info;
+                                break;
+                            }
+                        }
+                    }
+
+                    // Now try to create a new culture info from this value
+                    //
+                    if (retVal == null)
+                    {
+                        try
+                        {
+                            retVal = new CultureInfo(text);
+                        }
+                        catch { }
+                    }
+
+                    // Finally, try to find a partial match
+                    //
+                    if (retVal == null)
+                    {
+                        text = text.ToLower(CultureInfo.CurrentCulture);
+                        IEnumerator e = _values.GetEnumerator();
+                        while (e.MoveNext())
+                        {
+                            CultureInfo info = (CultureInfo)e.Current;
+                            if (info != null && GetCultureName(info).ToLower(CultureInfo.CurrentCulture).StartsWith(text))
```

Comment:
Rather than allocating new strings with ToLower, this can use StartsWith(string, StringComparison)